ReactApp1/ReactApp1.Server/Controllers/Domain/SharedSearchesController.cs

```diff
@@ -0,0 +1,41 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ReactApp1.Server.Extensions;
+using ReactApp1.Server.Data;
+using ReactApp1.Server.Models.Models.Base;
+
+namespace ReactApp1.Server.Controllers.Domain
+{
+    [ApiController]
+    [Route("api/")]
+    public class SharedSearchesController : ControllerBase
+    {
+        private readonly AppDbContext _context;
+        
+        public SharedSearchesController(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        [HttpGet("AllItems")]
+        public async Task<IActionResult> AllItems([FromQuery] string? search)
+        {
+            // TODO: Prideti i "Items" lenta FK EstablishmentId, kad pagal tai butu galima atrinkti tik tuos items, kurie priklauso tam establishmentui
+            var establishmentId = User.GetUserEstablishmentId();
+            if (establishmentId == null)
+            {
+                return BadRequest("Establishment not found.");
+            }
+
+            var result = await _context.Items
```

Comment:
Gal vertėtų šitą dalį perkelt į repository, nes čia kaip ir yra interakcija su duombaze?

Reply:
Aha, padariau repository